### ISSUE TITLE
Dialog title is marked as presentation element

### ISSUE THREAD

**Issue by mlewand** [ckeditor/ckeditor-dev#1408]
## Are you reporting a feature request or a bug?

Bug

## Provide detailed reproduction steps (if any)

If you look at our dialog title, it's marked as `[role=presentation]` element, thus might be inaccessible by some Screen Readers - which is a problem, because it's referenced as a dialog title by `[aria-labelledby]` attribute.

It works OK in JAWS, but will not work in screen readers that do not put presentation elements in Accessibility Tree.

To reproduce you could use Edge + Narrator, and follow the steps:

1. Open [CKEditor sample](http://nightly.ckeditor.com/latest/full/ckeditor/samples/).
2. Open link dialog.

### Expected result

The screen reader should mark dialog by saying:

> Dialog **Link** focused on (...)

### Actual result

Screen reader misses "Link" title and says:

> Dialog focused on (...)

**Comment by Comandeer** [comment]
It seems more like an upstream issue in Narrator. Specification for computing accessible name clearly states that even hidden elements could be used as accessible names if they are referenced by `[aria-labelledby]` attribute ([Section 4.3 Text Alternative Computation](https://w3c.github.io/accname/#mapping_additional_nd_te), comment for point 2.A):

> By default, assistive technologies do not relay hidden information, but an author can explicitly override that and include hidden text as part of the accessible name or accessible description by using `aria-labelledby` or `aria-describedby`.

`[role=presentation]` should be considered hidden for self-closing elements (e.g. `img`). As [hidden definition](https://w3c.github.io/accname/#dfn-hidden) states:

> Indicates that the element is not visible, perceivable, or interactive to _any_ user. An element is considered hidden if it or any one of its ancestor elements is not rendered or is explicitly hidden.

However for not self-closing elements [they should be treated as `span`](https://w3c.github.io/aria/#presentation):

> For any element with a role of presentation and which is not focusable, the user agent MUST NOT expose the implicit native semantics of the element (the role and its states and properties) to accessibility APIs. However, the user agent MUST expose content and descendant elements that do not have an explicit or inherited role of presentation. Thus, the presentation role causes a given element to be treated as having no role or to be removed from the accessibility tree, but does not cause the content contained within the element to be removed from the accessibility tree.

Therefore our label *should* be read by Narrator as it satisfy requirements for accessible names.

**Comment by mlewand** [comment]
Thanks for sharing @Comandeer, you made a perfectly valid point. In this case it's a third party issue and should be fixed in Narrator itself to respect the specification.
